save-common/src/commonMain/kotlin/org/cqfn/save/core/plugin/PluginConfig.kt

```diff
@@ -4,27 +4,25 @@
 
 package org.cqfn.save.core.plugin
 
+import org.cqfn.save.core.config.TestConfigSections
+
 import kotlinx.serialization.Serializable
 
 /**
- * Core interface for plugin configuration
+ * Core interface for plugin configuration (like warnPlugin/fixPLuin/e.t.c)
  */
-interface PluginConfig<T : PluginConfig<T>> {
+@Suppress("INLINE_CLASS_CAN_BE_USED")
+interface PluginConfig {
     /**
-     * Merge current config into [childConfig]
-     *
-     * @param childConfig list of child configs, which will be filtered for merging
+     * type of the config (usually related to the class: WARN/FIX/e.t.c)
      */
-    @Suppress("TYPE_ALIAS")
-    fun mergeConfigInto(childConfig: MutableList<PluginConfig<*>>)
+    val type: TestConfigSections
 
     /**
-     * Function which is capable for merging inherited configurations
-     *
-     * @param parentConfig config which will be merged with [this]
-     * @return corresponding merged config
+     * @param otherConfig - 'this' will be merged with 'other'
+     * @return merged config
      */
-    fun mergePluginConfig(parentConfig: T): T
+    fun mergeWith(otherConfig: PluginConfig): PluginConfig
```

Comment:
But this way we can't guarantee, that `otherConfig` is of the same class. Do you suggest to make checks inside plugins, or will it be like a contract?

Reply:
I thouht about pluses and minuses of that. Pluses:
- simple code 
- composition design pattern used

Minuses:
- we can face exceptions in Plugin code, **but** it will be easily caught on the first run as this method is called from single place where we have an explicit type check. **What could have gone wrong?** 👯‍♂️